appgate/secrets.py

```diff
@@ -126,10 +135,78 @@ def isinstance(value: PasswordField) -> bool:
         return isinstance(value, dict) and value.get("type") == "k8s/secret"
 
 
+class AppgateVault:
+    def __init__(self, client: Client):
+        self.client = client
+
+    def authenticate(self):
+        if self.client.is_authenticated():
```

Comment:
The naming is a bit confusing here, does `is_authenticated` do the actual authentication?

Reply:
For Kubernetes authentication (`AppgateInClusterVault` class), we call the login method to perform the authentication. 

For token authentication, (`AppgateExternalVault` class), it looks like the fact of possessing a token means that the user is authenticated. `is_authenticated()` here is a check to see if the provided token is actually valid - it's not actually performing a login. 

I was a bit confused by this too but reading their code seems to suggest that any authentication method that requires a explicit `login()` is an optional auth method listed in https://developer.hashicorp.com/vault/docs/auth. For example, if we wanted to support username/password authentication, vault admin needs to enable userpass auth method, then provided us with the username/password via env.